brave/src/main/java/brave/propagation/TraceContextOrSamplingFlags.java

```diff
@@ -81,64 +176,130 @@ public TraceContextOrSamplingFlags sampled(@Nullable Boolean sampled) {
     int flags = InternalPropagation.sampled(sampled, value.flags);
     if (flags == value.flags) return this; // save effort if no change
     return withFlags(flags);
   }
 
+  /**
+   * Returns non-{@code null} when both a {@linkplain TraceContext#traceIdString() trace ID} and
+   * {@linkplain TraceContext#spanIdString() span ID} were  {@link Extractor#extract(Object)
+   * extracted} from a request.
+   *
+   * <p>For example, given the header "b3: 80f198ee56343ba864fe8b2a57d3eff7-e457b5a2e4d86bd1-1",
+   * {@link B3Propagation} extracts the following:
+   * <ul>
+   *   <li>{@link TraceContext#traceIdString()}: "80f198ee56343ba864fe8b2a57d3eff7"</li>
+   *   <li>{@link TraceContext#spanIdString()}: "e457b5a2e4d86bd1"</li>
+   *   <li>{@link TraceContext#sampled()}: {@code true}</li>
+   * </ul>
+   *
+   * @return the trace context when {@link #traceIdContext()} ()} and {@link #samplingFlags()} are
+   * not {@code null}
+   * @see #create(TraceContext)
+   * @see #newBuilder(TraceContext)
+   * @since 4.0
+   */
   @Nullable public TraceContext context() {
     return type == 1 ? (TraceContext) value : null;
   }
 
+  /**
+   * Returns non-{@code null} when a {@linkplain TraceIdContext#traceIdString() trace ID} was {@link
+   * Extractor#extract(Object) extracted} from a request, but a {@linkplain
+   * TraceContext#spanIdString() span ID} was not.
+   *
+   * <p>For example, given the header "x-amzn-trace-id: Root=1-5759e988-bd862e3fe1be46a994272793",
+   * <a href="https://github.com/openzipkin/zipkin-aws/tree/master/brave-propagation-aws">AWSPropagation</a>
+   * extracts the following:
+   * <ul>
+   *   <li>{@link TraceIdContext#traceIdString()}: "15759e988bd862e3fe1be46a994272793"</li>
+   *   <li>{@link TraceIdContext#sampled()}: {@code null}</li>
+   * </ul>
+   *
+   * @return the trace ID context when {@link #context()} and {@link #samplingFlags()} are not
+   * {@code null}
+   * @see #create(TraceIdContext)
+   * @see #newBuilder(TraceIdContext)
+   * @since 4.9
+   */
   @Nullable public TraceIdContext traceIdContext() {
     return type == 2 ? (TraceIdContext) value : null;
   }
 
+  /**
+   * Returns non-{@code null} when a {@linkplain TraceContext#traceIdString() trace ID} was not
+   * {@link Extractor#extract(Object) extracted} from a request.
+   *
+   * <p>For example, given the header "b3: 1", {@link B3Propagation} extracts {@link #SAMPLED}.
+   *
+   * @return sampling flags when {@link #context()} and {@link #traceIdContext()} are not {@code
+   * null}
+   * @see #create(SamplingFlags)
+   * @see #newBuilder(SamplingFlags)
+   * @since 4.9
+   */
   @Nullable public SamplingFlags samplingFlags() {
     return type == 3 ? value : null;
   }
 
   /**
-   * Non-empty when {@link #context} is null: A list of additional state extracted from the
-   * request.
+   * Non-empty when {@link #context()} is {@code null}: A list of additional state extracted from
```

Comment:
See this is the same as before, but reformat to our normal grammar while we're here?

`Returns a list of additional state extracted from the request. Will be non-empty when {@link #context()} is {@code null}.`

Reply:
sg